Retrofit response issue resolved,RecyclerView  event Handler Gson Date formatter added,batch sql added

Aelafseged opened the pull request:
RecyclerView click and long click handler for detail display and edit/update
Date serialization for n/w, added with Gson date formatter.
Batch add onto local sqlite after fetching from n/w with onconflict method
Retrofit response issue resolved on empty body.

Aelafseged committed:
Added Retrofit response issue on empty body response.
RecyclerView click and long click handler for detail display and edit/update
Date serialization for n/w, added with Gson date formatter.
Batch add onto local sqlite after fetching from n/w with onconflict method

Aelafseged commented:
what date format shud we use for date_created/date_deleted?
and the primary key product_id... does it need to be auto_increment or server generated or user initiated?
cheers ;-)

Aelafseged committed:
Created Angular AfroSell Client with one custom component product-list component that fetch list of products using Observable from service that fetch data from the "product" spring server using address "http:localhost:3200/products"
**Took care of "Access-Control-Allow-Origin" header on the spring project by adding @CrossOrigin(origins = "*") annotation was a day long debugging...  after compilation it worked.
Used Intellij community edition

Aelafseged commented:
Have added the AfroSellDeskClient with edited product server side code

davewish committed:
eceptions  classes

davewish committed:
Merge pull request #6 from EriHackathon/product-dawit

eceptions  classes

EriHackathon commented on AfroSellAndroidClient/app/src/main/java/com/erihackton/shopping/source/remote/AfroSellRemoteDataSource.java:
```diff
@@ -21,7 +23,7 @@
 public class AfroSellRemoteDataSource implements DataSource {
     public static final String TAG = "AfroSellRemote";
-    public static final String baseUrl = "http://192.168.43.114:8000";
+    public static final String baseUrl = "http://192.168.43.114:3300";
```
Aelaf, why do we need to change this port I didn't see any update in the server side actually.

Aelafseged replied:
initially i was using my node.js server as a mock server, which was at port 8000 as to test my android app... now i'm using product spring server it's at 3300... but i'm continously changing the ip address..  when i'm using a internet cafe router that has dynamic ip and my phone hotspot

EriHackathon commented on product/src/main/java/com/erihackton/shopping/exceptions/ProductExceptionController.java:
```diff
@@ -0,0 +1,18 @@
+			
+    public ResponseEntity<Object> exception(ProductNotFoundException ex){
+		return new ResponseEntity<>("Product Not FOUND ",HttpStatus.NOT_FOUND);
+	
```
Dawit, I can see you have created an Exception class, but I don't see it where you are using it.

Aelafseged committed:
**On Android*
Added product View, Add, Edit and Delete functionality
with issue of newly created product Id on server  not reflecting on client side being an issue for offline repository.
**On DeskClient**
using angular bootstrap UI,
Added product View, Add functionality with Delete functionality not reflecting on UI side immediately.

Aelafseged committed:
Merge remote-tracking branch 'origin/ClientSideCRUD' into ClientSideCRUD

Aelafseged committed:
workshop clean up commit

Aelafseged committed:
Server and UI in sync on Delete action

Aelafseged committed:
Edit Product function completed with a glitch on Add/Editbutton on Angular part

Aelafseged committed:
clean up on compile code

Aelafseged committed:
merged origin/master with branch ClientSideCRUD

Aelafseged committed:
Merge branch 'ClientSideCRUD' into ClientSideCRUD2

Aelafseged committed:
CRUD+ operation and other expermentation branch

Aelafseged committed:
fixed Angular Edit PARTIALLY

Aelafseged committed:
fixed Angular Edit Fully using Service object as intent object

Aelafseged committed:
CRUD operations,material UI instead of bootstrap, basic error handling, reactive form, routing function

Aelafseged committed:
CRUD operations,material UI instead of bootstrap, basic error handling, reactive form, routing function
